In [1]:
import pandas as pd
import numpy as np
import datetime as dt 

In [2]:
df = pd.read_json('a3cleanedonly2015.json')
# df.head(20)

In [3]:
len(df['UID'].unique())

1537

In [4]:
# Number of races 

print(df['Race'].unique())
df['Race'].value_counts()


['Asian' 'Black' 'Hispanic' 'Other' 'Native' 'White' '']


White       751
Black       378
Hispanic    246
            114
Asian        19
Native       19
Other        10
Name: Race, dtype: int64

Data cleaning: Dropped column 'Unamed: 0' and 'FIELD1'. Using UID as the unique identifier. 
We have 1537 data points. 

114 missing points for Race (7% of the data). We converted it to 'Unknown'.

In [5]:
df.drop(columns = ['FIELD1', 'Unnamed: 0'], inplace = True)

In [6]:
df['Race'] = np.where(df['Race'] == '', 'Unknown', df['Race'])
df['Race'].value_counts()

White       751
Black       378
Hispanic    246
Unknown     114
Asian        19
Native       19
Other        10
Name: Race, dtype: int64

In [12]:
df.head(10)

UID                     Name   Age Gender   Race       Date  \
0   9978              Tim Elliott  53.0   Male  Asian 2015-01-02   
1  10077               Hung Trieu   NaN   Male  Asian 2015-02-01   
2  10374                 Lue Vang  39.0   Male  Asian 2015-04-22   
3  10421            Thong Kien Ma  32.0   Male  Asian 2015-05-05   
4  10628               Ton Nguyen  60.0   Male  Asian 2015-07-02   
5  10912             William Chau  59.0   Male  Asian 2015-09-09   
6  10918                  Tian Ma  31.0   Male  Asian 2015-09-10   
7  11009                Kevin Lau  44.0   Male  Asian 2015-10-11   
8  11166               Tuan Hoang   NaN   Male  Asian 2015-11-30   
9  11219  Mharloun Verdejo Saycon  39.0   Male  Asian 2015-12-14   

               City State Manner_of_death  Armed  Mental_illness   Flee  \
0           Shelton    WA            Shot                   True  False   
1           Houston    TX            Shot                  False  False   
2  Eldorado Springs    CO            Shot                  False  False   
3    South El Monte    CA            Shot                  False  False   
4         San Diego    CA            Shot                  False  False   
5          El Monte    CA            Shot  Knife           False  False   
6           Potsdam    NY            Shot                  False  False   
7      Lake Charles    LA            Shot    Gun           False  False   
8            Aurora    CO            Shot                  False  False   
9        Long Beach    CA            Shot  Knife           False  False   

       Month  
0    January  
1   February  
2      April  
3        May  
4       July  
5  September  
6  September  
7    October  
8   November  
9   December

In [13]:
# Grouping data by month 

# Extract month from Date 

df['Month'] = df['Date'].dt.month

# df

In [14]:
# Group by month 
monthly_shootings = df.groupby(['Month', 'Race'])['UID'].count().reset_index()
monthly_shootings.rename(columns = {'UID' : 'Count'}, inplace = True)
monthly_shootings

Month      Race  Count
0       1     Asian      4
1       1     Black     26
2       1  Hispanic     21
3       1    Native      3
4       1   Unknown      6
..    ...       ...    ...
63     12     Black     31
64     12  Hispanic     23
65     12     Other      4
66     12   Unknown      9
67     12     White     57

[68 rows x 3 columns]

In [15]:
month_and_race = monthly_shootings.pivot(index='Month', columns='Race', values='Count')
month_and_race

Race   Asian  Black  Hispanic  Native  Other  Unknown  White
Month                                                       
1        4.0   26.0      21.0     3.0    NaN      6.0   58.0
2        1.0   29.0      19.0     NaN    NaN      9.0   59.0
3        2.0   42.0      18.0     3.0    NaN     11.0   74.0
4        2.0   41.0      22.0     3.0    NaN      7.0   64.0
5        1.0   31.0       8.0     NaN    2.0     17.0   67.0
6        NaN   23.0      24.0     1.0    NaN      6.0   49.0
7        1.0   31.0      25.0     2.0    2.0      8.0   83.0
8        NaN   33.0      23.0     NaN    NaN     15.0   65.0
9        2.0   29.0      20.0     7.0    1.0     12.0   59.0
10       1.0   35.0      23.0     NaN    NaN     14.0   54.0
11       1.0   27.0      20.0     NaN    1.0      NaN   62.0
12       4.0   31.0      23.0     NaN    4.0      9.0   57.0

In [16]:
#month_and_race['Month'] = pd.to_datetime(month_and_race['Month'], format='%m').dt.month_name()

KeyError: 'Month'

In [ ]:
month_and_race.to_csv('/Users/salma/Documents/GitHub/CAPP30239-FA22/week_06/police_hw/month_and_race.csv')

In [ ]:
test = pd.read_csv('/Users/salma/Documents/GitHub/CAPP30239-FA22/week_06/police_hw/month_and_race.csv')
# test

In [ ]:
# GRAPH 2:Shootings per month 

by_month = df.groupby('Month')['UID'].count().reset_index()
by_month.rename(columns = {'UID' : 'Count'}, inplace = True)
by_month


In [ ]:
by_month.to_csv('/Users/salma/Documents/GitHub/CAPP30239-FA22/week_06/police_hw/monthly_shootings.csv', index = False)


In [ ]:
test = pd.read_csv('/Users/salma/Documents/GitHub/CAPP30239-FA22/week_06/police_hw/month_and_race.csv')
test